huaweicloud/services/waf/resource_huaweicloud_waf_instance_group_associate.go

```diff
@@ -0,0 +1,169 @@
+		UpdateContext: resourceWafInsGroupAssociateUpdate,
+		DeleteContext: resourceWafInsGroupAssociateDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"load_balances": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceWafInsGroupAssociateCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.WafDedicatedV1Client(conf.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating HuaweiCloud WAF dedicated client : %s", err)
+	}
+
+	groupID := d.Get("group_id").(string)
+	group, err := pools.Get(client, groupID)
+	if err != nil {
+		return fmtp.DiagErrorf("Error querying WAF instance group: %s", err)
+	}
+
+	if len(group.Bindings) > 0 {
+		// Remove the bound instance
+		for _, v := range group.Bindings {
+			err = pools.RemoveELB(client, groupID, v.ID)
+			if err != nil {
+				return fmtp.DiagErrorf("Error removing load balance[%s] from the group[%s]: %s", v, groupID, err)
+			}
+		}
+	}
+
+	elbIDs := d.Get("load_balances").(*schema.Set).List()
+	mErr := addELBInstances(client, groupID, elbIDs)
+	if mErr != nil && mErr.ErrorOrNil() != nil {
+		return diag.FromErr(mErr)
+	}
+	d.SetId(groupID)
+
+	return resourceWafInsGroupAssociateRead(ctx, d, meta)
+}
+
+func addELBInstances(c *golangsdk.ServiceClient, groupID string, ids []interface{}) *multierror.Error {
+	var mErr *multierror.Error
+	for _, v := range ids {
+		lbID := v.(string)
+		_, e := pools.AddELB(c, groupID, lbID)
+		if e != nil {
+			err := fmtp.Errorf("Error in binding load balance[%s] to the group[%s]: %s", lbID, groupID, e)
+			mErr = multierror.Append(mErr, err)
+		}
+	}
+	return mErr
+}
+
+func resourceWafInsGroupAssociateRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.WafDedicatedV1Client(conf.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating HuaweiCloud WAF dedicated client : %s", err)
+	}
+
+	group, err := pools.Get(client, d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "Error querying WAF instance group")
+	}
+
+	loadBalances := make([]interface{}, 0, len(group.Bindings))
+	for _, v := range group.Bindings {
+		loadBalances = append(loadBalances, v.Name)
+	}
+	mErr := multierror.Append(nil,
+		d.Set("group_id", group.ID),
+		d.Set("load_balances", loadBalances),
+	)
+
+	if err = mErr.ErrorOrNil(); err != nil {
+		return fmtp.DiagErrorf("error setting WAF dedicated group attributes: %s", err)
```

Comment:
Maybe `if mErr.ErrorOrNil() != nil` and `return fmtp.DiagErrorf("...: %s", mErr)` just ok.

Reply:
nice